From 89add5d7700e4ee4743c9b6e4fe17af48ea34c06 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Tue, 9 Dec 2014 16:55:12 +0800
Subject: [PATCH] config: board: change "Bus Speed" choice appearance's
 dependence.

Change to depend on the board's decision, instead of the "CPU clock" selection.

Because one CPU clock will export several bus speeds,
but some bus speeds which aren't used on the board make user confused.

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
Acked-by: Nicolas Ferre <nicolas.ferre@atmel.com>
---
 board/Config.in                        | 40 +++++++++++++++++++++++++++-------
 board/at91sam9260ek/Config.in.board    |  1 +
 board/at91sam9261ek/Config.in.board    |  1 +
 board/at91sam9263ek/Config.in.board    |  1 +
 board/at91sam9g10ek/Config.in.board    |  1 +
 board/at91sam9g20ek/Config.in.board    |  1 +
 board/at91sam9m10g45ek/Config.in.board |  1 +
 board/at91sam9n12ek/Config.in.board    |  1 +
 board/at91sam9rlek/Config.in.board     |  1 +
 board/at91sam9x5ek/Config.in.board     |  1 +
 board/sama5d3_xplained/Config.in.board |  2 ++
 board/sama5d3xek/Config.in.board       |  3 +++
 board/sama5d4_xplained/Config.in.board |  1 +
 board/sama5d4ek/Config.in.board        |  1 +
 14 files changed, 48 insertions(+), 8 deletions(-)

diff --git a/board/Config.in b/board/Config.in
index 86d3bde..6852863 100644
--- a/board/Config.in
+++ b/board/Config.in
@@ -278,38 +278,62 @@ choice
 
 config CONFIG_BUS_SPEED_83MHZ
 	bool "83 MHz"
-	depends on CONFIG_CPU_CLK_166MHZ
+	depends on SUPPORT_BUS_SPEED_83MHZ
 
 config CONFIG_BUS_SPEED_90MHZ
 	bool "90 MHz"
-	depends on ALLOW_CPU_CLK_180MHZ
+	depends on SUPPORT_BUS_SPEED_90MHZ
 
 config CONFIG_BUS_SPEED_100MHZ
 	bool "100 MHz"
-	depends on CONFIG_CPU_CLK_200MHZ || CONFIG_CPU_CLK_400MHZ
+	depends on SUPPORT_BUS_SPEED_100MHZ
 
 config CONFIG_BUS_SPEED_133MHZ
 	bool "133 MHz"
-	depends on CONFIG_CPU_CLK_528MHZ || CONFIG_CPU_CLK_400MHZ || CONFIG_CPU_CLK_396MHZ || CONFIG_CPU_CLK_266MHZ
+	depends on SUPPORT_BUS_SPEED_133MHZ
 
 config CONFIG_BUS_SPEED_148MHZ
 	bool "148 MHz"
-	depends on CONFIG_CPU_CLK_444MHZ
+	depends on SUPPORT_BUS_SPEED_148MHZ
 
 config CONFIG_BUS_SPEED_166MHZ
 	bool "166 MHz"
-	depends on CONFIG_CPU_CLK_498MHZ || CONFIG_CPU_CLK_332MHZ
+	depends on SUPPORT_BUS_SPEED_166MHZ
 
 config CONFIG_BUS_SPEED_170MHZ
 	bool "170 MHz"
-	depends on CONFIG_CPU_CLK_510MHZ
+	depends on SUPPORT_BUS_SPEED_170MHZ
 
 config CONFIG_BUS_SPEED_176MHZ
 	bool "176 MHz"
-	depends on CONFIG_CPU_CLK_528MHZ && !SAMA5D3X
+	depends on SUPPORT_BUS_SPEED_176MHZ
 
 endchoice
 
+config SUPPORT_BUS_SPEED_83MHZ
+	bool
+
+config SUPPORT_BUS_SPEED_90MHZ
+	bool
+
+config SUPPORT_BUS_SPEED_100MHZ
+	bool
+
+config SUPPORT_BUS_SPEED_133MHZ
+	bool
+
+config SUPPORT_BUS_SPEED_148MHZ
+	bool
+
+config SUPPORT_BUS_SPEED_166MHZ
+	bool
+
+config SUPPORT_BUS_SPEED_170MHZ
+	bool
+
+config SUPPORT_BUS_SPEED_176MHZ
+	bool
+
 config CONFIG_TRUSTZONE
 	bool
 	default n
diff --git a/board/at91sam9260ek/Config.in.board b/board/at91sam9260ek/Config.in.board
index fb0e2e0..88c8fa8 100644
--- a/board/at91sam9260ek/Config.in.board
+++ b/board/at91sam9260ek/Config.in.board
@@ -11,5 +11,6 @@ config	CONFIG_AT91SAM9260EK
 	select ALLOW_BOOT_FROM_DATAFLASH_CS1
 	select ALLOW_DATAFLASH_RECOVERY
 	select ALLOW_NANDFLASH_RECOVERY
+	select SUPPORT_BUS_SPEED_100MHZ
 	help
 	    Use the AT91SAM9260EK Development board
diff --git a/board/at91sam9261ek/Config.in.board b/board/at91sam9261ek/Config.in.board
index 8fad9e5..85d4188 100644
--- a/board/at91sam9261ek/Config.in.board
+++ b/board/at91sam9261ek/Config.in.board
@@ -11,5 +11,6 @@ config	CONFIG_AT91SAM9261EK
 	select ALLOW_BOOT_FROM_DATAFLASH_CS3
 	select ALLOW_DATAFLASH_RECOVERY
 	select ALLOW_NANDFLASH_RECOVERY
+	select SUPPORT_BUS_SPEED_100MHZ
 	help
 	    Use the AT91SAM9261EK Development board
diff --git a/board/at91sam9263ek/Config.in.board b/board/at91sam9263ek/Config.in.board
index a15d5cc..a1b239e 100644
--- a/board/at91sam9263ek/Config.in.board
+++ b/board/at91sam9263ek/Config.in.board
@@ -15,5 +15,6 @@ config	CONFIG_AT91SAM9263EK
 	select ALLOW_BOOT_FROM_DATAFLASH_CS0
 	select ALLOW_DATAFLASH_RECOVERY
 	select ALLOW_NANDFLASH_RECOVERY
+	select SUPPORT_BUS_SPEED_100MHZ
 	help
 	    Use the AT91SAM9263EK Development board
diff --git a/board/at91sam9g10ek/Config.in.board b/board/at91sam9g10ek/Config.in.board
index cb0881c..d146256 100644
--- a/board/at91sam9g10ek/Config.in.board
+++ b/board/at91sam9g10ek/Config.in.board
@@ -12,5 +12,6 @@ config	CONFIG_AT91SAM9G10EK
 	select ALLOW_BOOT_FROM_DATAFLASH_CS3
 	select ALLOW_DATAFLASH_RECOVERY
 	select ALLOW_NANDFLASH_RECOVERY
+	select SUPPORT_BUS_SPEED_133MHZ
 	help
 	    Use the AT91SAM9G10EK Development board
diff --git a/board/at91sam9g20ek/Config.in.board b/board/at91sam9g20ek/Config.in.board
index 5e5df15..5774a6f 100644
--- a/board/at91sam9g20ek/Config.in.board
+++ b/board/at91sam9g20ek/Config.in.board
@@ -11,5 +11,6 @@ config	CONFIG_AT91SAM9G20EK
 	select ALLOW_BOOT_FROM_DATAFLASH_CS1
 	select ALLOW_DATAFLASH_RECOVERY
 	select ALLOW_NANDFLASH_RECOVERY
+	select SUPPORT_BUS_SPEED_133MHZ
 	help
 	    Use the AT91SAM9G20EK Development board
diff --git a/board/at91sam9m10g45ek/Config.in.board b/board/at91sam9m10g45ek/Config.in.board
index 4a87c0d..4cc5fec 100644
--- a/board/at91sam9m10g45ek/Config.in.board
+++ b/board/at91sam9m10g45ek/Config.in.board
@@ -11,6 +11,7 @@ config	CONFIG_AT91SAM9M10G45EK
 	select ALLOW_BOOT_FROM_DATAFLASH_CS0
 	select ALLOW_DATAFLASH_RECOVERY
 	select ALLOW_NANDFLASH_RECOVERY
+	select SUPPORT_BUS_SPEED_133MHZ
 	help
 	    Use the AT91SAM9M10G45EK Development board
 	    Can also be used for AT91SAM9G45/AT91SAM9M10 family
diff --git a/board/at91sam9n12ek/Config.in.board b/board/at91sam9n12ek/Config.in.board
index 789d205..617aacf 100644
--- a/board/at91sam9n12ek/Config.in.board
+++ b/board/at91sam9n12ek/Config.in.board
@@ -13,5 +13,6 @@ config	CONFIG_AT91SAM9N12EK
 	select ALLOW_NANDFLASH_RECOVERY
 	select ALLOW_PIO3
 	select CPU_HAS_PMECC
+	select SUPPORT_BUS_SPEED_133MHZ
 	help
 	    Use the AT91SAM9N12EK Development board
diff --git a/board/at91sam9rlek/Config.in.board b/board/at91sam9rlek/Config.in.board
index 1a13834..94d1413 100644
--- a/board/at91sam9rlek/Config.in.board
+++ b/board/at91sam9rlek/Config.in.board
@@ -11,5 +11,6 @@ config	CONFIG_AT91SAM9RLEK
 	select ALLOW_BOOT_FROM_DATAFLASH_CS0
 	select ALLOW_DATAFLASH_RECOVERY
 	select ALLOW_NANDFLASH_RECOVERY
+	select SUPPORT_BUS_SPEED_100MHZ
 	help
 	    Use the AT91SAM9RLEK Development board
diff --git a/board/at91sam9x5ek/Config.in.board b/board/at91sam9x5ek/Config.in.board
index c36c4b2..ebe197b 100644
--- a/board/at91sam9x5ek/Config.in.board
+++ b/board/at91sam9x5ek/Config.in.board
@@ -14,5 +14,6 @@ config	CONFIG_AT91SAM9X5EK
 	select ALLOW_PIO3
 	select CPU_HAS_PMECC
 	select CONFIG_HAS_HW_INFO
+	select SUPPORT_BUS_SPEED_133MHZ
 	help
 	    Use the AT91SAM9X5EK Development board
diff --git a/board/sama5d3_xplained/Config.in.board b/board/sama5d3_xplained/Config.in.board
index cb465c4..634ce00 100644
--- a/board/sama5d3_xplained/Config.in.board
+++ b/board/sama5d3_xplained/Config.in.board
@@ -16,5 +16,7 @@ config CONFIG_SAMA5D3_XPLAINED
 	select CONFIG_MMC_SUPPORT
 	select CONFIG_PM
 	select CONFIG_HAS_PMIC_ACT8865
+	select SUPPORT_BUS_SPEED_133MHZ
+	select SUPPORT_BUS_SPEED_166MHZ
 	help
 	    Use the SAMA5D3 Xplained development board
diff --git a/board/sama5d3xek/Config.in.board b/board/sama5d3xek/Config.in.board
index cd783be..4cf23f1 100644
--- a/board/sama5d3xek/Config.in.board
+++ b/board/sama5d3xek/Config.in.board
@@ -19,5 +19,8 @@ config CONFIG_SAMA5D3XEK
 	select CPU_HAS_PMECC
 	select CONFIG_HAS_HW_INFO
 	select CONFIG_MMC_SUPPORT
+	select SUPPORT_BUS_SPEED_133MHZ
+	select SUPPORT_BUS_SPEED_148MHZ
+	select SUPPORT_BUS_SPEED_166MHZ
 	help
 	    Use the SAMA5D3EK Development board
diff --git a/board/sama5d4_xplained/Config.in.board b/board/sama5d4_xplained/Config.in.board
index 964ae52..683facf 100644
--- a/board/sama5d4_xplained/Config.in.board
+++ b/board/sama5d4_xplained/Config.in.board
@@ -20,5 +20,6 @@ config CONFIG_SAMA5D4_XPLAINED
 	select CONFIG_TRUSTZONE
 	select CONFIG_HAS_PMIC_ACT8865
 	select CONFIG_PM
+	select SUPPORT_BUS_SPEED_176MHZ
 	help
 	  Use the SAMA5D4_Xplained Development board
diff --git a/board/sama5d4ek/Config.in.board b/board/sama5d4ek/Config.in.board
index 57e9cff..b5e4b16 100644
--- a/board/sama5d4ek/Config.in.board
+++ b/board/sama5d4ek/Config.in.board
@@ -20,5 +20,6 @@ config CONFIG_SAMA5D4EK
 	select CONFIG_TRUSTZONE
 	select CONFIG_HAS_PMIC_ACT8865
 	select CONFIG_PM
+	select SUPPORT_BUS_SPEED_176MHZ
 	help
 	  Use the SAMA5D4EK Development board
-- 
1.9.1

